packages/site/cms-site/src/graphQLFetch/graphQLFetch.ts

```diff
@@ -30,9 +30,32 @@ function graphQLHeaders(previewData?: SitePreviewData) {
     return headers;
 }
 
-//from graphql-request https://github.com/jasonkuhrt/graphql-request/blob/main/src/raw/functions/gql.ts
+// Adapted from https://github.com/jasonkuhrt/graffle/blob/main/src/layers/6_helpers/gql.ts
 export const gql = (chunks: TemplateStringsArray, ...variables: unknown[]): string => {
```

Comment:
here some suggestion to improve the code.

`export const gql = (chunks: TemplateStringsArray, ...variables: unknown[]): string => {
    const generateErrorMessage = (chunk: string): string => {
        let message =
            `Non-string variable in the GraphQL document\n\n` +
            `This is most likely due to importing a GraphQL document from a React Client Component.\n` +
            `All GraphQL documents need to be imported from React Server Components (i.e. no "use client" notation).`;

        if (chunk.trim().length > 0) {
            message += `\n\nThe error occurred in the following GraphQL document:\n${chunk}`;
        }

        return message;
    };

    return chunks.reduce((acc, chunk, index) => {
        const variable = variables[index];

        if (index in variables && typeof variable !== "string") {
            throw new Error(generateErrorMessage(chunk));
        }

        return `${acc}${chunk}${variable || ""}`;
    }, "");
};
`

Reply:
How would this "improve" the code? 😉 IMO readability takes a hit when implementing your changes.